### PR TITLE
chore: add connection route

### PR DESCRIPTION
**Pull request by subirjolly**
Add a connection route so that we can provide it `email` parameter and retrieve its corresponding SSO connection.

### CODE REVIEW

File: src/unity.yml [hunk 1]
```diff
@@ -61,6 +61,8 @@ paths:
     $ref: './unity/paths/operator_orgs_orgId.yml'
   '/operator/orgs/{orgId}/limits':
     $ref: './unity/paths/operator_orgs_orgId_limits.yml'
+  '/api/v2private/auth/connection':
```
Comment (Asalem1) [inline src/unity.yml:64]: this should be to the `api/v2/quartz` path

**Review by Asalem1 — CHANGES_REQUESTED** [review]

File: src/unity.yml (hunk 1 above)
Comment (Asalem1) [reply src/unity.yml:64]: Actually, sorry, this should just be `/auth/connection`

File: src/unity.yml [hunk 2]
```diff
@@ -61,6 +61,8 @@ paths:
     $ref: './unity/paths/operator_orgs_orgId.yml'
   '/operator/orgs/{orgId}/limits':
     $ref: './unity/paths/operator_orgs_orgId_limits.yml'
+  '/api/v2/quartz/auth/connection':
```
Comment (Asalem1) [inline src/unity.yml:64]:
```suggestion
  '/auth/connection':
```

File: contracts/priv/unity.yml [hunk 3]
```diff
@@ -966,6 +966,34 @@ paths:
         default:
           description: Unexpected error
           $ref: '#/components/responses/ServerError'
+  /api/v2/quartz/auth/connection:
+    get:
+      operationId: GetConnection
+      tags:
+        - Connection
+      parameters:
+        - $ref: '#/components/parameters/TraceSpan'
```
Comment (randycoulman) [inline contracts/priv/unity.yml:975]: For my own enlightenment: What does this line do?

File: contracts/priv/unity.yml [hunk 4]
```diff
@@ -966,6 +966,34 @@ paths:
...
+          name: email
+          schema:
+            type: string
+            format: email
+          required: true
+          description: The email to get connection for.
+      responses:
+        '200':
+          content:
+            text/plain:
```
Comment (randycoulman) [inline contracts/priv/unity.yml:986]: I think we're going to return this as `application/json`; not sure if that will matter to the generated client code.

**Review by randycoulman — APPROVED** [review]
LGTM, though I think the response type might be wrong. I'll approve, but please double-check that before merging.

File: contracts/priv/unity.yml (hunk 3 above)
Comment (subirjolly) [reply contracts/priv/unity.yml:975]: I am not sure what TraceSpan does but I think we pass some meta info in headers through this.

File: contracts/priv/unity.yml (hunk 4 above)
Comment (subirjolly) [reply contracts/priv/unity.yml:986]: ah. what's the structure going to be for this response?

Comment (randycoulman) [reply contracts/priv/unity.yml:986]: A JSON-encoded string. See https://github.com/influxdata/quartz/pull/4799 for an example curl request.

**Comment by randycoulman** [comment]
I just remembered one more thing: this endpoint doesn't require any auth credentials; not sure if that needs to be called out somehow here.  We need to be able to make this call prior to a user being logged in, so there's no way to authenticate it using our usual mechanisms.

**Review by Asalem1 — APPROVED** [review]

File: contracts/priv/unity.yml (hunk 3 above)
Comment (subirjolly) [reply contracts/priv/unity.yml:975]: I don't think there's any way to specify if a route is authenticated.

**Pull request merged**